src/store/portfolioSummary/index.ts

```diff
@@ -0,0 +1,256 @@
+   * the response and sets the costs to the respective clin.
+   */
+  @Action({rawError: true})
+  private async setCostsToTaskOrderClins(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    const clinNumbers: string[] = [];
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinNumbers.push(clinRecord.clin_number);
+        })
+      })
+    });
+    const allCostList = await api.costsTable.getQuery(
+      {
+        params:
+          {
+            sysparm_fields: 'sys_id,clin,task_order_number,is_actual,value',
+            sysparm_query: `clinIN` + clinNumbers
+          }
+      }
+    )
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinRecord.cost_records =
+            // @ts-ignore
+            allCostList.filter(cost => cost.clin === clinRecord.clin_number &&
+              cost.task_order_number === taskOrder.task_order_number); // FIXME temp code
+          // allCostList.filter(cost => cost.clin?.value === clinRecord.sys_id);//FIXME correct code
+        })
+      })
+    })
+  }
+
+  /**
+   * Performs the 'Total Obligated' and 'Funds Spent' for all the portfolios in the list
+   *
+   * NOTE: Instead of computing here if we were to make the aggregate API call, it would take
+   * as many calls as the number of portfolios to get, for example, the 'Total Obligated' from
+   * the CLINS table. The reason for this is, the aggregate query response does not have any
+   * reference to what portfolio the aggregate belongs to and so a call for each portfolio.
+   * Computing here will eliminate all these calls.
+   */
+  @Action({rawError: true})
+  private computeAllAggregations(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.dod_component = 'ARMY' // FIXME: delete this line after API starts returning
+      let totalObligatedForPortfolio = 0;
+      let totalFundsSpentForPortfolio = 0;
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          if (clinRecord.clin_status === 'ACTIVE' ||
+            clinRecord.clin_status === 'OPTION_EXERCISED') { // TODO: double check the statuses
+            totalObligatedForPortfolio =
+              totalObligatedForPortfolio + Number(clinRecord.funds_obligated);
+          }
+          clinRecord.cost_records?.forEach(costRecord => {
+            totalFundsSpentForPortfolio = totalFundsSpentForPortfolio + Number(costRecord.value);
+          })
+        })
+      })
+      portfolio.funds_obligated = totalObligatedForPortfolio;
+      portfolio.funds_spent = totalFundsSpentForPortfolio;
+    })
+  }
+
+  @Action({rawError: true})
+  public async loadPortfolioSummaryList(loggedInUserId: string): Promise<PortfolioSummaryDTO[]> {
```

Comment:
not sure we'd need to pass the `loggedInUserId` as a param to this method. Once we establish the current user object in a vuex store somewhere, we'll be able to just reach out there to get it.

Reply:
Removed the parameter and adjusted the comment.